### PR TITLE
refactor(app): clean up redux server state management

### PR DESCRIPTION
**Pull request by vabruzzo**
# Overview
Removes parts of redux server state management that we do not use, including `modules`, `robot`, & `support`. Fixes small bugs caused by incorrect redux store data derivations. This partially cleans up some of the redux directory as a step towards migrating all server state management to ReactQuery. Because this PR has grown large, and because some of the deletions require surgical care, we will do this in multiple tickets.

Closes RAUT-126.

# Changelog

- removes `modules` redux state except for `updateModule` action which is still used with `dispatchApiRequest`
- removes legacy `robot` state. this includes `session`, `calibration`, and `connection`, and parts of other redux domains that relied on what was removed here.
- adds `isRunRunning` property to `useRunStatuses` to fix bug caused by now deleted `getIsRunning` `robot` selector. the selector was deriving state from a part of the store that we no longer updated, the bug this led to was sometimes broken button disabled states
- cleans up a bunch of `analytics` redux code that was not being used, particularly code that depended on other non-analytics redux code deleted in this PR, some code was commented and not deleted to keep for reference, similar to what we did for other events.

# Review requests
- general smoke testing, this PR deletes a lot of code

# Risk assessment
High.

### CODE REVIEW

File: app/src/organisms/Devices/HeaterShakerWizard/__tests__/TestShake.test.tsx [hunk 1]
```diff
@@ -138,6 +138,7 @@ describe('TestShake', () => {
       isLatchClosed: true,
     } as any)
     mockUseRunStatuses.mockReturnValue({
+      isRunRunning: false,
```
Comment (vabruzzo) [inline app/src/organisms/Devices/HeaterShakerWizard/__tests__/TestShake.test.tsx:141]: this replaces the `redux/robot/session` `getIsRunning` selector. this slice of redux state was not being populated so it was a silent bug in 2 spots where we were incorrectly deriving a button's disabled state

File: app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/FactoryResetModal.tsx [hunk 2]
```diff
@@ -31,14 +31,14 @@ import type { ResetConfigRequest } from '../../../../../redux/robot-admin/types'
 
 interface FactoryResetModalProps {
   closeModal: () => void
-  isRobotConnected: boolean
```
Comment (vabruzzo) [inline app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/FactoryResetModal.tsx:32]: We don't have this notion of connected to a robot anymore. Changed name accordingly.

File: app/src/organisms/Devices/RobotSettings/CalibrationDetails/OverflowMenu.tsx [hunk 3]
```diff
@@ -88,7 +88,7 @@ export function OverflowMenu({
     calBlockModalState,
     setCalBlockModalState,
   ] = React.useState<CalBlockModalState>(CAL_BLOCK_MODAL_CLOSED)
-  const isRunning = useSelector(getIsRunning)
```
Comment (vabruzzo) [inline app/src/organisms/Devices/RobotSettings/CalibrationDetails/OverflowMenu.tsx:91]: This was a subtle bug, this slice of state was never updated so the selector never returned true. A button was being rendered and it's `disabled` state was partially determined by this broken `isRunning` selector.

File: app/src/organisms/Devices/RobotSettings/RobotSettingsAdvanced.tsx [hunk 4]
```diff
@@ -73,10 +73,10 @@ export function RobotSettingsAdvanced({
   const settings = useSelector<State, RobotSettings>((state: State) =>
     getRobotSettings(state, robotName)
   )
-  const connected = robot?.connected != null && robot.connected
```
Comment (vabruzzo) [inline app/src/organisms/Devices/RobotSettings/RobotSettingsAdvanced.tsx:76]: This was a bug. The connected state was never being updated. Further down the component tree the value of this state was being updated according to `robot?.status !== UNREACHABLE`, so I've done that here.

File: app/src/redux/analytics/__tests__/make-event.test.ts [hunk 5]
```diff
@@ -32,78 +26,13 @@ const getAnalyticsPipetteCalibrationData = selectors.getAnalyticsPipetteCalibrat
...
-const getProtocolAnalyticsData = selectors.getProtocolAnalyticsData as jest.MockedFunction<
-  typeof selectors.getProtocolAnalyticsData
->
 
 describe('analytics events map', () => {
   beforeEach(() => {
     jest.resetAllMocks()
   })
 
-  it('robot:CONNECT -> robotConnected event', () => {
```
Comment (vabruzzo) [inline app/src/redux/analytics/__tests__/make-event.test.ts:44]: I am fairly certain that this event should no longer be firing.

File: app/src/redux/analytics/__tests__/make-event.test.ts [hunk 6]
```diff
@@ -32,78 +26,13 @@ const getAnalyticsPipetteCalibrationData = selectors.getAnalyticsPipetteCalibrat
...
-  })
-
   describe('events with protocol data', () => {
-    const protocolData = { foo: 'bar' } as any
-
-    beforeEach(() => {
-      getProtocolAnalyticsData.mockResolvedValue(protocolData)
-    })
-
-    it('robot:PROTOCOL_LOAD > protocolUploadRequest', () => {
```
Comment (vabruzzo) [inline app/src/redux/analytics/__tests__/make-event.test.ts:97]: im pretty sure this event should also no longer before firing, but if not we need to re-add it. it definitely is not firing in 6.0

File: app/src/redux/analytics/selectors.ts [hunk 7]
```diff
@@ -186,114 +160,114 @@ export function getAnalyticsPipetteCalibrationData(
...
+  //     calibrationExists: Boolean(pipcal),
+  //     markedBad: pipcal?.status?.markedBad ?? false,
+  //     // @ts-expect-error protect for cases where model is not on pip
+  //     pipetteModel: pip.model,
+  //   }
+  // }
   return null
 }
 
 export function getAnalyticsTipLengthCalibrationData(
```
Comment (b-cooper) [inline app/src/redux/analytics/selectors.ts:179]: These functions are still called when the legacy calibration sessions are launched, it seems like they're probably not reporting any params now that we can't get the connected robot from the removed selector.  This code should be updated to have the robot name passed in from the epic level. I know that won't be tackled in this PR, but I want to make sure we don't delete this stuff, instead we should add a TODO comment and address the broken analytics in a following ticket

Comment (vabruzzo) [reply app/src/redux/analytics/selectors.ts:179]: @b-cooper TODOs added with reference to the created ticket RAUT-150. :+1:

**Review by b-cooper — APPROVED** [review]
✂️

**Pull request merged**